File: LeetCode167.cpp
#include<vector>
#include<numeric>
#include<iostream>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int low = 0, high = numbers.size() - 1;
        while (low < high) {
            int sum = numbers[low] + numbers[high];
            if (sum == target) {
                return {low + 1, high + 1};
            } else if (sum < target) {
                ++low;
            } else {
                --high;
            }
        }
        return {-1, -1};
    }
};

int main(){
    vector<int> numbers;
    int n=0,t=0;
    cout << "number of nums: ";
    cin >> n;
    for (int i = 1; i <= n; i++){
        cin >> t;
        numbers.push_back(t);
    }
    cout << "number of target: ";
    cin >> n;
    Solution sol;
    numbers = sol.twoSum(numbers, n);
    cout << "[" << numbers[0] << "," << numbers[1] << "]";
    return 0;
}
/**
 * 输入：numbers = [2,7,11,15], target = 9
 * 输出：[1,2]
 * 解释：2 与 7 之和等于目标数 9 。因此 index1 = 1, index2 = 2 。
 */

File: LeetCode167.java
import java.util.Scanner;

public class LeetCode167{
    public static void main(String[] args) {
        LeetCode167 solution = new LeetCode167();
        Scanner scanner = new Scanner(System.in);
        int[] numbers;
        int n = 0;
        System.out.print("number of numbers: ");
        n = scanner.nextInt();
        numbers = new int[n];
        for (int i = 1; i <= n; i++){
            numbers[i-1] = scanner.nextInt();
        }
        System.out.print("target: ");
        n = scanner.nextInt();
        solution.twoSum(numbers, n);
        scanner.close();
    }

    public int[] twoSum(int[] numbers, int target) {
        int low = 0, high = numbers.length - 1;
        while (low < high) {
            int sum = numbers[low] + numbers[high];
            if (sum == target) {
                return new int[]{low + 1, high + 1};
            } else if (sum < target) {
                ++low;
            } else {
                --high;
            }
        }
        return new int[]{-1, -1};
    }
}
// number of numbers:4
// 2 7 11 13
// target:9